ynput/OpenPype
# Anatomy: Preparation for different root overrides

## Brief description
Prepare Anatomy to handle only 'studio' site override on it's own.

## Description
Change how Anatomy fill root overrides based on requested site name. The logic which decide what is active site was moved to sync server addon and the same for receiving root overrides of local site. The Anatomy resolve only studio site overrides anything else is handled by sync server. BaseAnatomy only expect root overrides value and does not need site name. Validation of site name happens in sync server same as resolving if site name is local or not.

## Additional information
We could consider to change `site_name` to `site_type` as in anatomy is technically possible to use only `studio` and `local` where `local` is not an id of site but type of site (but passing the logic to sync server technically gives the option).
Goal is to move sync server responsibility of resolving root overrides to sync server and keep studio root overrides in Anatomy to prepare the logic for AYON overrides where they will be separated that way.

## Testing notes:
- Anatomy should work as before
- The initialization of anatomy should be fast as before (or faster) -> especially on creating multiple instances of Anatomy
- Local root overrides from Local settings should be applied properly in Anatomy object

Root overrides without sync server
1. Disabled Sync server
2. Restart process
3. Go to local settings
4. Change roots in a project
5. Create anatomy object (e.g. in Tray > Admin > Console)
6. Validate root value if is applied properly (keep in mind it is cached for 10 seconds so it may take time to propagate)

Root overrides with sync server
1. Enable Sync server
2. Restart process
3. Change settings and change root
- active site can be changed in project settings and in local settings
- roots can be changed only in local settings
5. Create anatomy object (e.g. in Tray > Admin > Console)
6. Validate root value if is applied properly (keep in mind it is cached for 10 seconds so it may take time to propagate)
